Fixes the issue where lib folder with file '_._' is ignored

deepakaravindr opened the pull request:
Fixes https://github.com/NuGet/Home/issues/1021.

The issue is that lib folder with just the file '_._' is not considered as
a valid 'any' framework. That is, lib/_._ is being ignored. However,
lib/net45/_._ or lib/net45 will be considered as valid.

Added a property to FrameworkSpecificGroup which says whether there was
an empty folder represented by file '_._'.

Even after this change, an empty folder of lib without the '_._' file
is not a valid empty directory and will be ignored.

@yishaigalatzer @emgarten @MeniZalzman @feiling

emgarten commented on src/NuGet.Core/NuGet.Packaging.Core.Types/FrameworkSpecificGroup.cs:
```diff
@@ -23,7 +23,10 @@ public class FrameworkSpecificGroup : IEquatable<FrameworkSpecificGroup>, IFrame
+        /// <param name="hasEmptyFolder">Sets hasEmptyFolder if not null. Otherwise, it is computed.</param>
+        public FrameworkSpecificGroup(NuGetFramework targetFramework,
+            IEnumerable<string> items,
+            bool? hasEmptyFolder = null)
```
Remove optional parameter

emgarten commented:
:rocket:

deepakaravindr committed:
Fixes https://github.com/NuGet/Home/issues/1021.

The issue is that lib folder with just the _._ file is not considered as
a valid 'any' framework. That is, lib/_._ is being ignored. However,
lib/net45/_._ or lib/net45 will be considered as valid.

Added a property to FrameworkSpecificGroup which says whether there was
an empty folder represented by file '_._'.

Even after this change, an empty folder of lib without the '_._' file
is not a valid empty directory and will be ignored.

emgarten commented:
:rocket: :rocket:

yishaigalatzer commented on src/NuGet.Core/NuGet.Packaging.Core.Types/FrameworkSpecificGroup.cs:
```diff
@@ -57,6 +59,11 @@ public IEnumerable<string> Items
             get { return _items; }
         }
 
+        public bool HasEmptyFolder
```
nit: oneline

yishaigalatzer commented on src/NuGet.Core/NuGet.ProjectManagement/Utility/MSBuildNuGetProjectSystemUtility.cs:
```diff
@@ -15,30 +15,52 @@ namespace NuGet.ProjectManagement
         {
-            return (frameworkSpecificGroup != null && frameworkSpecificGroup.Items != null &&
-                    (frameworkSpecificGroup.Items.Any() || !frameworkSpecificGroup.TargetFramework.Equals(NuGetFramework.AnyFramework)));
+            return (frameworkSpecificGroup != null
```
put this in a readable if, and return true/false

yishaigalatzer commented:
:shipit: if @emgarten is fine with it

deepakaravindr committed:
Incorporated comments

deepakaravindr commented:
Merged with https://github.com/NuGet/NuGet.Client/commit/da45244508288d40b58429dee0bbb44ce172d804 and 
https://github.com/NuGet/NuGet.Client/commit/a188fa5df8efe38a4a1bac6fdf0af352bf63edf3
